src/components/TodoAppContent/TodoAppContent.tsx

```diff
@@ -0,0 +1,237 @@
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useRef,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  editTodo,
+} from '../../api/todos';
+import { useAuth } from '../../Context/Context';
+import { TodoItem } from '../TodoItem';
+import { Filter } from '../Filter';
+import { filteredData } from '../../helpers/filteredData';
+import { FilterBy, ErrorMessage, EditField } from '../../types/types';
+import { Error } from '../Error';
+
+export const TodoAppContent: React.FC = () => {
+  const userId = useAuth();
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodoField, setNewTodoField] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [todosCount, setTodosCount] = useState(0);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadTodosIds, setLoadTodosIds] = useState<number[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [disabledClear, setDisabledClear] = useState(true);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const currentTodos = filteredData<Todo>(todos, filterBy);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const countOfNotCompleted = todos
+    .filter(({ completed }) => !completed);
+
+  const countOfCompleted = todos.filter(({ completed }) => completed);
+
+  useEffect(() => {
+    if (userId) {
+      getTodos(userId)
+        .then((resp) => {
+          setTodos(resp);
+        })
+        .catch(() => setErrorMessage(ErrorMessage.Load));
+    }
+  }, [userId]);
+
+  useEffect(() => {
+    setTodosCount(countOfNotCompleted.length);
+  }, [todos]);
```

Comment:
pay attention to warning and paste missing dependency to array

Reply:
Here is no missing dependency, countOfNotCompleted dependency only of changeng todos. Adding dependency: 'countOfNotCompleted.length' will lead to infinity loop. We can ignore this warning, if we sure in dependencies we use